docs/source/configuration.rst

```diff
@@ -66,6 +66,42 @@ When tokens are received from an issuer during redemption,
 these are the only public keys which will satisfy the redeemer and cause the tokens to be made available to the client to be spent.
 Tokens received with any other public key will be sequestered and will *not* be spent until some further action is taken.
 
+lease.crawl-interval.mean
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This item controls the frequency at which the lease maintenance crawler runs.
+The lease maintenance crawler visits all shares and renews their leases if necessary.
+The crawler will run at random intervals.
+The client will try to make the average (mean) interval between runs equal to this setting.
+The value is an ISO8601 duration string.
```

Comment:
I don't think I've ever encountered this format, so I wondered a bit about the use of it here. Looking at the examples here and the discussion [on wikipedia](https://en.wikipedia.org/wiki/ISO_8601#Durations), the format looks reasonable, and readable (once you ignore the leading `P`).

It does have an alternative format (which is like `P0000-00-26T00:00:00`), which is looks like isodate has bunch of bugs in parsing (and is unmaintained). There is also isoduartion (which was inspired by iso date), but it also has a bunch of bugs in parsing the alternative format.

I've gone back-and-forth a bit on whether it makes sense to use this format:
- My first thought was that this was a weird thing, and maybe we shouldn't use it. I was going to point out that [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339) is the format used in practice for dates which is a restricted profile of ISO 8601, though people often refer to that format as ISO 8601; and that the other bits of ISO 8601 are not commonly encountered.
- After thinking about it a bit more, and looking at the wikipedia page for the spec, I decided that it seemed like a reasonable format; it the regular form is relatively unambiguous[1] and mostly human readable. Thus, while a bit odd, it seemed like a reasonable choice.
- However, when I came back to look at it again, I got curious about the alternate format, and went to look at how the parsing for it was implemented. Both isodate and isoduration defer to a date parse for the alternative format, which is incorrect[2]. Given the state of parsers, I'm leaning away from using it; The appeal of using a standardized format like this is to re-use existing code, but the quality of the implementations I can find don't hold up.

[1] Though as pointed out on the wikipedia page, `P1M` could be ambiguous.
[2] It at least accepts too much, since it accepts any date format (isodate uses its own, isoduration uses arrow's), which both accept more than what is defined for ISO 8061. I think they may accept to little (though the copies of ISO 8061 I could find aren't clear on this), as they don't let you have have 0 for any of the components.

Reply:
Did you have any ideas for an alternative format to use?

Reply:
Given that we aren't exposing this value to users, we could probably get away with something simple like number of seconds or days. Slightly more complex, we could allow suffixes for days/hours/minutes/seconds.

Reply:
Oh yea.  Number of seconds seems fine.  Thanks.